packages/crypto/lib/src/crypto/aes.rs

```diff
@@ -22,35 +24,41 @@ pub struct AES {
 #[wasm_bindgen]
 impl AES {
     #[wasm_bindgen(constructor)]
-    pub fn new(key: Vec<u8>, iv: Vec<u8>) -> Result<AES, String> {
-        if key.len() < 32 {
-            return Err(format!("{} Received {}", AESError::KeyLengthError, key.len()));
+    pub fn new(key: VecU8Pointer, iv: Vec<u8>) -> Result<AES, String> {
+        if key.length < 32 {
+            return Err(format!("{} Received {}", AESError::KeyLengthError, key.length));
         }
-        let key = GenericArray::from_iter(key.into_iter());
+        let mut key = GenericArray::from_iter(key.vec.clone().into_iter());
         let iv: [u8; 12] = match iv.try_into() {
             Ok(iv) => iv,
-            Err(_) => return Err(AESError::IVSizeError.to_string()),
+            Err(_) => {
+                key.zeroize();
+                return Err(AESError::IVSizeError.to_string())
+            }
         };
 
-        Ok(AES {
+        let aes = AES {
             cipher: Aes256Gcm::new(&key),
             iv,
-        })
+        };
+        key.zeroize();
+        Ok(aes)
     }
 
-    pub fn encrypt(&self, text: &str) -> Result<Vec<u8>, String> {
+    pub fn encrypt(&self, mut text: String) -> Result<Vec<u8>, String> {
```

Comment:
Not sure about the change from `&str -> String` in the `text` parameter here, but I think it's right because we want to zeroize the text after we've received it so the method should gobble up the string. wasm-bindgen seems to handle both the same anyway.